### PR TITLE
Change CoreLib native image to be R2R by default on all platforms

### PR DESCRIPTION
**Pull request by jkotas**
This change is part of larger workstream to reconfigure code generation for .NET Core 3.0. Other key parts of this effort are using IBC to optimize the framework size, and turning on tiered JIT by default.

In .NET Core, we are using R2R native images for everything except CoreLib on x86 and x64. This change is switching CoreLib on x86 and x64 too. The expected 
- We will have one native image format to focus on and optimize for.
- The CoreLib native image will be smaller (together with the IBC change, CoreLib went from 8,494,080 down to 3,973,120 in our testing)
- The native code in CoreLib may be slightly worse. This will be complensated for by tiered JIT. (Note that the degradation in CoreLib is much less than what we see everywhere else because of there are no cross-module inlining concerns in CoreLib.)

### CODE REVIEW

File: src/tools/crossgen/crossgen.cpp [hunk 1]
```diff
@@ -835,12 +833,6 @@ int _cdecl wmain(int argc, __in_ecount(argc) WCHAR **argv)
     // Are we compiling mscorlib.dll? 
     bool fCompilingMscorlib = StringEndsWith((LPWSTR)pwzFilename, CoreLibName_IL_W);
```
Comment (nattress) [inline src/tools/crossgen/crossgen.cpp]: `fCompilingMscorlib` looks like it's not useful anymore. It's passed through to `ComputeTPAListFromPlatformAssembliesPath` and then on to `PopulateTPAList` neither of which do anything useful with it.

**Comment by AndyAyersMS** [comment]
@jkotas Can you add the rationale for this change to the PR, and some discussion of the expected impact?

**Comment by jkotas** [comment]
> @jkotas Can you add the rationale for this change to the PR, and some discussion of the expected impact?

Done.

**Comment by AndyAyersMS** [comment]
Looks like we see as much as a 10% slowdown in tiering for some of the jit-intensive scenario tests (blue circle ... ignore the red circled point at it was a noise spike), impacting both tiered and non-tiered runs:

r2rimpact

Seems likely this is because we need to jit more methods but there should be some extra runtime cost in there too.

I don't think we measure pure runtime startup time (eg time to main for hello world or something similar) where we could minimize jit contribution. Perhaps we should. @jorive something to consider...

cc @kouvel @noahfalk

**Comment by jkotas** [comment]
@AndyAyersMS Thanks for heads up. I will take a look what we can do about this. 400ms for hello world is a lot either way.

**Comment by noahfalk** [comment]
Thanks for raising this @AndyAyersMS  and for taking a look @jkotas ! I like the direction of these changes, but perhaps we should defer it a bit? My concern is that @kouvel is still trying to land tiered compilation and I imagine as soon as he does there will be a wave of feedback as it spreads into scenarios for the first time. It would be helpful to ensure the impact of default tiered compilation can be analyzed independently from the impact of this.

**Comment by jkotas** [comment]
I would think we should use 2.2 previews to evaluate the tied JIT impact in isolation.

For 3.0, we have number of related changes in flight. https://github.com/dotnet/coreclr/pull/19046 is another one. The simple thing is to get all of them in and then start getting feedback on all them do together as soon as possible. If you would like to stage them, that's fine with me too - but it needs a plan.

**Pull request merged**